blocks/utils.py

```diff
@@ -429,3 +429,47 @@ def ipdb_breakpoint(x):
     """
     import ipdb
     ipdb.set_trace()
+
+
+class LambdaIterator(six.Iterator):
```

Comment:
The thing when one says `WhateverIterator` it just feels like an iterator over a sequence of whatevers. Let it be `LambdaIterator` until we come up with something significantly better.